src/util/arith_tools.cpp

```diff
@@ -281,10 +283,31 @@ void mp_max(mp_integer &a, const mp_integer &b)
 /// \param bit_index: index (0 is the least significant)
 bool get_bitvector_bit(const irep_idt &src, std::size_t bit_index)
 {
-  // The representation is binary, using '0'/'1',
-  // most significant bit first.
-  PRECONDITION(bit_index < src.size());
-  return src[src.size() - 1 - bit_index] == '1';
+  // The representation is hex, most significant nibble first.
+  const auto nibble_index = bit_index >> 2;
+
+  if(nibble_index >= src.size())
+    return false;
+
+  char nibble = src[src.size() - 1 - nibble_index];
+
+  DATA_INVARIANT(isxdigit(nibble), "");
```

Comment:
Invariants shouldn't have an empty description string.

Reply:
Still to be addressed.